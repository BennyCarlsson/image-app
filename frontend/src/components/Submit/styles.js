import styled from 'styled-components'

export const StyledInput = styled.input`
  line-height: 14px;
  box-shadow: 0 1px 2px 0 rgb(0 0 0 / 10%);
  text-decoration: none;
  background-color: #607bff;
  color: #fff;
  font-family: Lato, sans-serif;
  font-weight: 700;
  font-size: 1.2rem;
  padding: 1.3rem 4rem;
  -webkit-appearance: none;
  appearance: none;
  border-radius: 25rem;
  border: none;
  margin-top: 64px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;

  :hover {
    background-color: #6099ff;
  }

  :disabled {
    opacity: 0.6;
    cursor: not-allowed;
    box-shadow: none;
  }
`
