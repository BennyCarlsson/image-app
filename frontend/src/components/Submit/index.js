import React from 'react'
import { StyledInput } from './styles'

const Submit = ({ disabled = false }) => {
  return <StyledInput type='submit' value='Submit' disabled={disabled} />
}

export default Submit
