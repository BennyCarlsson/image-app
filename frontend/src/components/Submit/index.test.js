import React from 'react'
import { render, screen } from '@testing-library/react'
import Submit from '.'

describe('<Submit />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Submit />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<Submit />)
    expect(screen).toBeTruthy()
  })

  it('should render button not disabled', () => {
    render(<Submit />)
    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(screen.getByRole('button')).not.toBeDisabled()
  })

  it('should render button disabled', () => {
    render(<Submit disabled />)
    expect(screen.getByRole('button')).toBeDisabled()
  })
})
