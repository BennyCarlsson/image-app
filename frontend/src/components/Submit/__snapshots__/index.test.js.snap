// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Submit /> should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-bdvvtL jAGkdz"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;
